plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCatalogConnectorSmokeTest.java

```diff
@@ -160,13 +172,256 @@ public void testCommentView()
         }
     }
 
+    @Test
+    public void testRegisterTableWithTableLocation()
+    {
+        TestIcebergRegisterTableProcedure.StorageFormat storageFormat = ORC;
+        String baseTableName = "test_register_table_with_table_location_" + storageFormat + "_" + randomTableSuffix();
+        String trinoTableName = trinoTableName(baseTableName);
+
+        assertUpdate(format("CREATE TABLE %s (a int, b varchar, c boolean) with (format = '%s')", trinoTableName, storageFormat));
+        assertUpdate(format("INSERT INTO %s values(1, 'INDIA', true)", trinoTableName), 1);
+        assertUpdate(format("INSERT INTO %s values(2, 'USA', false)", trinoTableName), 1);
+        assertThat(query(format("SELECT * FROM %s", trinoTableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')");
+
+        String tableLocation = getTableLocation(trinoTableName);
+        String showCreateTableOld = (String) computeActual("SHOW CREATE TABLE " + trinoTableName).getOnlyValue();
+
+        // Drop table from glue metastore and use the same table name to register again with the metadata
+        dropTableFromMetastore(baseTableName);
+
+        assertUpdate(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", schemaName, baseTableName, tableLocation));
+        String showCreateTableNew = (String) computeActual("SHOW CREATE TABLE " + trinoTableName).getOnlyValue();
+
+        assertUpdate(format("INSERT INTO %s values(3, 'POLAND', true)", trinoTableName), 1);
+        assertThat(query(format("SELECT * FROM %s", trinoTableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')," +
+                        " ROW(INT '3', VARCHAR 'POLAND', BOOLEAN 'true')");
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew);
+        assertUpdate(format("DROP TABLE %s", trinoTableName));
+    }
+
+    @Test
+    public void testRegisterTableWithMetadataFile()
+    {
+        // TODO find the way to use filesystem api to get the latest metadata and then enable the test
```

Comment:
```
TODO find the way to use filesystem api to get the latest metadata and then enable the test
```

The test is already enabled.

Reply:
Modified the TODO comment. 

What is the better way to have a test method which needs to be TODO? When I was commenting `@Test` annotation, IDE was throwing exception.